controllers/nifiuser_controller.go

```diff
@@ -173,60 +173,61 @@ func (r *NifiUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 		pkiManager := pki.GetPKIManager(r.Client, cluster)
 
-		// check if marked for deletion
+		// check if marked for deletion. Otherwise, reconcile the certificate
 		if k8sutil.IsMarkedForDeletion(instance.ObjectMeta) {
 			r.Log.Info("Nifi user is marked for deletion, revoking certificates and removing finalizers.",
 				zap.String("user", instance.Name))
 			if err = pkiManager.FinalizeUserCertificate(ctx, instance); err != nil {
 				return RequeueWithError(r.Log, "failed to finalize certificate for user "+instance.Name, err)
 			}
-		}
-
```

Comment:
This looks like a large change, but it's just wrapping this block of code in an `else`. That is, only reconcile certificates if the `NifiUser` is not marked for deletion.